src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/longjump/modes/VulcanLongjump.kt

```diff
@@ -0,0 +1,122 @@
+ *
+ *
+ */
+
+package net.ccbluex.liquidbounce.features.module.modules.movement.longjump.modes
+
+import net.ccbluex.liquidbounce.config.Choice
+import net.ccbluex.liquidbounce.config.ChoiceConfigurable
+import net.ccbluex.liquidbounce.event.events.PacketEvent
+import net.ccbluex.liquidbounce.event.events.PlayerMoveEvent
+import net.ccbluex.liquidbounce.event.handler
+import net.ccbluex.liquidbounce.event.repeatable
+import net.ccbluex.liquidbounce.features.module.ModuleManager
+import net.ccbluex.liquidbounce.features.module.modules.movement.longjump.ModuleLongJump
+import net.ccbluex.liquidbounce.utils.entity.moving
+import net.ccbluex.liquidbounce.utils.entity.sqrtSpeed
+import net.ccbluex.liquidbounce.utils.entity.strafe
+import net.ccbluex.liquidbounce.utils.movement.zeroXZ
+import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket
+import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket.Full
+import net.minecraft.network.packet.s2c.play.PlayerPositionLookS2CPacket
+
+/**
+ * @anticheat Vulcan
+ * @anticheatVersion 2.8.9
+ * @testedOn anticheat-test.com, eu.loyisa.cn
+ */
+internal object VulcanLongjump : Choice("Vulcan") {
+    override val parent: ChoiceConfigurable<*>
+        get() = ModuleLongJump.mode
+    var recievedLagback = false
+    var started = false
+
+    override fun enable() {
+        recievedLagback = false
+        started = false
+        ModuleLongJump.jumped = false
+        ModuleLongJump.boosted = false
+    }
+
+    val positions = listOf(
+        0.41999998688698,
+        0.7531999805212,
+        1.00133597911214,
+        1.16610926093821,
+        1.24918707874468,
+        1.25220334025373,
+        1.17675927506424,
+        1.02442408821369,
+        0.79673560066871,
+        0.49520087700593,
+        0.1212968405392,
+        0.0
+    )
+
+    val repeatable = repeatable {
+
+        if (started && player.hurtTime == 10) {
+            player.setPosition(player.pos.x, player.pos.y - 0.5, player.pos.z)
+        }
+        if (started && recievedLagback) {
+            player.velocity.y = 1.0
+            player.setPosition(player.pos.x, player.pos.y + 8, player.pos.z)
+            player.strafe(strength = 1.0, speed = 4.2)
+            recievedLagback = false
+        }
+        if (started && player.hurtTime == 5) {
+            player.setPosition(player.pos.x, player.pos.y + 8, player.pos.z)
+            player.strafe(strength = 1.0, speed = 0.3)
+            started = false
+            ModuleLongJump.jumped = true
+            ModuleLongJump.boosted = true
+        }
+        if(player.age % 2 == 0) {
+            player.velocity.y = -0.0971
+        } else {
+            player.velocity.y = -0.148
+        }
+        if (player.isOnGround && !recievedLagback && player.hurtTime == 0) {
+            if(!(ModuleLongJump.autoDisable && ModuleLongJump.jumped)) {
+                repeat(3) {
+                    for (position in positions) {
```

Comment:
As for this one, declaring `private val repeatedList = List(3) { positions }.flatten()` straight after the 'positions' declaration and iterating through the repeated list instead (`for (position in repeatedList) {`) can improve the code readability a bit :)
Although, it will cost extra memory.

Reply:
I have decided to skip this as it doesnt make the code much clearer